File: kattis_dicegame.hs
main :: IO()
main = do
    a <- getLine
    b <- getLine
    let x = sum (map read (words a))
    let y = sum (map read (words b))
    printMax x y

printMax :: (Ord a) => a -> a -> IO()
printMax x y
    | x > y     = putStrLn "Gunnar"
    | x < y     = putStrLn "Emma"
    | otherwise = putStrLn "Tie"

File: kattis_dicegame.cpp
#include <iostream>

using namespace std;

int main()
{
    int a, b, c, d, gunnar, emma;

    cin >> a >> b >> c >> d;
    gunnar = a + b + c + d;

    cin >> a >> b >> c >> d;
    emma = a + b + c + d;

    if (gunnar > emma)
    {
        cout << "Gunnar" << endl;
    }
    else if (emma > gunnar)
    {
        cout << "Emma" << endl;
    }
    else
    {
        cout << "Tie" << endl;
    }
}
